release.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -e
+
+current_branch=$(git branch --show-current)
+git_changes=$(git status -s | wc -l)
+if [[ $git_changes != 0 ]]; then
+	if [[ $current_branch == "master" ]]; then
+		echo "You have modified the master branch."
+		echo "Please revert or move your changes to another branch before running this script."
+	else
+		echo "Please commit or stash your changes before running this script"
+	fi
+	exit 1
+fi
+
+git fetch
+ahead=$(git status -sb | grep ahead -c)
+if [[ $ahead != 0 ]]; then
+	echo "Your local master branch is ahead of the remote master branch. Exiting."
+	exit 1
+fi
+
+
+release_type=$1
+if [[ $release_type != "minor" && $release_type != "patch" && $release_type != "major" ]]; then
+	echo "Invalid argument: $release_type"
+	exit 1
+fi
+
+read -ep "Are you sure you want to cut a $release_type release? [y/N] " continue
+if [[ $continue != "yes" && $continue != "y" && $continue != "Yes" && $continue != "YES" ]]; then
+	exit
+fi
+
+git checkout master
```

Comment:
It seems like we assume we're on master, based on the error messages earlier in the script. I don't think the release script should be doing anything to the working tree (acknowledging that this script currently has to make a branch). Either we're on `master` or a release series branch, or we shouldn't work at all.

Reply:
Damn, I thought I moved that line further up in the script. I must have copied it over before I made that edit. I'll fix that.

Reply:
This script should never make a branch. For reasons to do with how the CI deployments work (that we can get into later), we are always pushing the version bump and the release tag directly to master. The intention of the checks at the beginning of the script was to account for human error, i.e. someone forgetting to commit or stash changes, or forgetting to checkout the master branch before running the script.

Reply:
I think you can just remove it entirely. Check to see if you're on `master` and if it's up to date with the remote, and in either error case, just bail with a helpful error message. The moment we try to get too clever, weird bugs start popping up.